airflow/serialization/serialized_objects.py

```diff
@@ -195,6 +196,19 @@ def serialize_to_json(
                 serialized_object[key] = value
         return serialized_object
 
+    @classmethod
+    def serialize_timezone(cls, var: datetime.tzinfo) -> Any:
+        """Serializes a timezone to json"""
+        tz_name = getattr(var, 'name', None)
+        if not tz_name:
+            try:
+                tz_name = var.tzname(datetime.datetime.utcnow())
+            except InvalidTimezone:
+                raise TypeError(f"the timezone {var} can't be serialized")
+
+
+        return cls._encode(str(tz_name), type_=DAT.TIMEZONE)
```

Comment:
`tzname()` can return `None` for naive datetimes, so we can’t just cast it to str. We either need to raise or do something about it.

Reply:
done!